vhdbuilder/scripts/automate_release.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+set -euo pipefail
+source vhdbuilder/scripts/automate_helpers.sh
+set +x
+
+EV2_ARTIFACT_PIPELINE_ID="319030" 
+SIG_RELEASE_PIPELINE_ID="494"
+
+RELEASE_EV2_ARTIFACTS_ALIAS_NAME="ev2_artifacts"
+RELEASE_API_ENDPOINT="https://msazure.vsrm.visualstudio.com/CloudNativeCompute/_apis/Release/releases?api-version=5.1"
+
+ADO_PAT="${ADO_PAT:-""}"
+VHD_BUILD_ID="${VHD_BUILD_ID:-""}"
+
+trigger_ev2_artifacts() {
+    echo "creating EV2 artifacts for VHD build with ID: $VHD_BUILD_ID"
+
+    RESPONSE=$(az pipelines run --id $EV2_ARTIFACT_PIPELINE_ID --variables "VHD_PIPELINE_RUN_ID=$VHD_BUILD_ID")
+    EV2_BUILD_ID=$(echo "$RESPONSE" | jq -r '.id')
+    EV2_BUILD_NUMBER=$(echo "$RESPONSE" | jq -r '.buildNumber')
+    STATUS="$(az pipelines runs show --id $EV2_BUILD_ID | jq -r '.status')"
+
+    while [ "${STATUS,,}" == "notstarted" ] || [ "${STATUS,,}" == "inprogress" ]; do
+        echo "EV2 artifact build $EV2_BUILD_ID is still in-progress..."
+        sleep 60
+        STATUS="$(az pipelines runs show --id $EV2_BUILD_ID | jq -r '.status')"
+    done
+
+    if [ "${STATUS,,}" != "completed" ]; then
+        echo "EV2 artifact build failed for VHD build with ID: $VHD_BUILD_ID, failed build ID: $EV2_BUILD_ID"
+        return 1
+    fi
+    echo "EV2 artifacts successfully built for VHD build with ID: $VHD_BUILD_ID, EV2 build ID: $EV2_BUILD_ID"
+}
+
+create_release() {
+    echo "creating SIG release for VHD with build ID: $VHD_BUILD_ID"
+
+    # we do it this way for now since the devops CLI extension doesn't support what we need: https://github.com/Azure/azure-devops-cli-extension/issues/1257
+    echo "sending POST request to $RELEASE_API_ENDPOINT"
+    REQUEST_BODY="{'artifacts': [{'alias': '$RELEASE_EV2_ARTIFACTS_ALIAS_NAME', 'instanceReference': {'id': '$EV2_BUILD_ID', 'name': '$EV2_BUILD_NUMBER'}}], 'definitionId': $SIG_RELEASE_PIPELINE_ID}"
+    RESPONSE=$(curl -X POST -H "Authorization: Basic $(echo -n ":$ADO_PAT" | base64)" -H "Content-Type: application/json" -d "$REQUEST_BODY" "$RELEASE_API_ENDPOINT")
+    
+    RELEASE_ID=$(echo "$RESPONSE" | jq -r '.id')
+    echo "SIG release successfully created for VHD build with ID: $VHD_BUILD_ID"
+    echo "release URL: https://msazure.visualstudio.com/CloudNativeCompute/_releaseProgress?_a=release-pipeline-progress&releaseId=$RELEASE_ID"
```

Comment:
nit: I wonder if its possible to parse out the release approval ev2 link as well from the API? that way, it reduces another click for the release manager.

Reply:
I like the way you think

Reply:
unfortunately it isn't included in the response JSON - I'll create a follow-up item to investigate if this is possible some other way